src/components/forms/NumberField/NumberField.js

```diff
@@ -0,0 +1,80 @@
+import {
+  FormFieldDescription,
+  FormLabel,
+  Paragraph,
+  Textbox,
+  FormField as UtrechtFormField,
+} from '@utrecht/component-library-react';
+import {useField} from 'formik';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {getBEMClassName} from 'utils';
+
+export const NumberField = ({
+  min,
+  step,
+  name = '',
+  label = '',
+  isRequired = false,
+  description = '',
+  id = '',
+  disabled = false,
+  invalid = false,
+}) => {
+  const labelClassName = getBEMClassName('label', [isRequired && 'required'].filter(Boolean));
+
+  const onChange = e => {
+    const inputValue = e.target.value;
+    const numericValue = Number(inputValue);
+
+    // Check if the input value is a valid number or a hyphen
+    if (!isNaN(numericValue) || inputValue === '-') {
+      // Check if the numeric value is below the minimum value
+      if (min && numericValue < min) {
+        return;
+      }
+      // Check if the numeric value is not divisible by the step value
+      if (step && (numericValue % step !== 0 || inputValue.includes('.'))) {
```

Comment:
additional thought - the input value should be interpreted according to the browser locale and be normalized as such :grimacing: 

e.g. NL/Dutch browser should be able to enter `1,5` while EN should enter `1.5` and this should both normalize to the float `1.5` in the actual form data.